simulate/simulate.go

```diff
@@ -229,7 +235,13 @@ func MutateGene(inputSeq []dna.Base, branchLength float64, geneFile string) []dn
 		}
 
 	}
-	finalSequence = BaseExtToBases(newSequence)
+	mutatedSequence := BaseExtToBases(newSequence)
+	delFound := seqContainsDels(mutatedSequence, dna.Gap)
+	if delFound == false && deletions == true {
+		finalSequence = deleteBase(mutatedSequence)
```

Comment:
I think this always puts in a deletion in the first branch in the tree.  Can we use branch length to control the likelihood of this happening on different branches?